backend2/src/core/article.py

```diff
@@ -159,18 +159,42 @@ def parse_new_article(self):
 
         self.db.commit()
 
+    def parse_all_new_articles(self):
+        articles = self.db.query(Article) \
+            .filter(Article.status == ArticleStatus.FETCHED) \
+            .with_for_update(skip_locked=True, key_share=True) \
+            .all()
```

Comment:
Please don't lock multiple entities in the database at once. It prevents this command from running in parallel manner. Moreover, locking all articles at once prevents any other activities with them including running internal database service tasks. If parsing goes wrong we'll have a bunch of articles stuck in locked state waiting for timeout.
See how [fetch_new_articles](https://github.com/CovidMutations/VirusMutationsAI/blob/master/backend2/src/commands/articles.py#L8) processes articles one by one.